### ISSUE TITLE
Empty heading error on a non empty h2

### ISSUE THREAD

**Issue by laudrain** [daisy/ace#170]
Please provide the following details if possible or relevant:

- [1.0.0] the version of Ace impacted by the issue (as returned by the `ace -v` command).
- [v8.7.0] the version of Node used to run Ace (as returned by the `node -v` command).
- [MacOS 10.13.3] your operating system and version.
- [] if the issue is a bug when running Ace: the error message as produced in the log file (you can enable debug logging with the `--verbose` option).
- [] if the issue is about the HTML report: the browser(s) and version(s) you used.
- [x] if possible, let us know if you can share a sample input file to help us reproduce the issue.

I don't understand why Ace gives an error "minor | Best Practice | empty-heading" on this document:

```
<body id="e-Accessible_Knowledge2015LAU-1" epub:type="frontmatter">
	<section class="Title-Page-Display" epub:type="titlepage">
		<h1 epub:type="title">e-Accessible<br/>Knowledge</h1>
		<h2 epub:type="subtitle" role="doc-subtitle" class="Title-Page-Subtitle">A G3ict Business Case White Paper Series</h2>
		<p class="Title-Page-Paragraph">Spring 2016</p>
	</section>
</body>
```
on 

```
Snippet:
<h2 xmlns="http://www.w3.org/1999/xhtml" xmlns:epub="http://www.idpf.org/2007/ops" epub:type="subtitle" role="doc-subtitle" class="Title-Page-Subtitle">A G3ict Business Case White Paper Series</h2>
```

log attached
[e-A.log](https://github.com/daisy/ace/files/1689671/e-A.log)

**Comment by rdeltour** [comment]
I could reproduce, it seems to be a bug.
Interestingly, the same doc passes when you remove the `epub:type` and `role` attributes…

**Comment by laudrain** [comment]
Yes. it is ok also when I replace h2 by p:
```
<section class="Title-Page-Display" epub:type="titlepage">
		<h1 epub:type="title">e-Accessible<br/>Knowledge</h1>
		<p epub:type="subtitle" role="doc-subtitle" class="Title-Page-Subtitle">A G3ict Business Case White Paper Series</p>
		<p class="Title-Page-Paragraph">Spring 2016</p>
</section>

```
The problem is that there no subhead tag in HTML.

**Comment by laudrain** [comment]
Still the same problem on 2 successive h1, one for title and the second for subtitle:
<section class="div_autre" epub:type="appendix" role="doc-appendix">
  <div class="appen" id="appen-005"><span epub:type="pagebreak" id="page_183" title="183" role="doc-pagebreak"/>
    **<h1 epub:type="title" class="bmtit">Le Séminaire</h1>
    <h1 epub:type="subtitle" class="bmstit" role="doc-subtitle">Ordre chronologique</h1>**

Ace violation message is inappropriate: empty-heading.

Workaround can be using p tag instead of the second h1:
  <div class="appen" id="appen-005"><span epub:type="pagebreak" id="page_183" title="183" role="doc-pagebreak"/>
    <h1 epub:type="title" class="bmtit">Le Séminaire</h1>
    <p epub:type="subtitle" class="bmstit" role="doc-subtitle">Ordre chronologique</p>

Any planning to fix this?

**Comment by rdeltour** [comment]
Sorry @laudrain for the very late action on this issue.

The problem comes from the definition of the `doc-subtitle` role in DPUB ARIA 1.0, which says the accessible name only comes from author defined attributes (e.g. `aria-label` or `aria-labelledby`) and not from the contents of the element.

This was filed as w3c/dpub-aria#6 and is fixed in DPUB 1.1 editor’s draft. 

I'll submit a patch to axe-core (and Ace) to proactively implement this fix.

**Comment by rdeltour** [comment]
Fixed with PR #184
